control-plane/src/modules/router.ts

```diff
@@ -695,4 +703,60 @@ export const router = s.router(contract, {
       body: await getObject(library),
     };
   },
+  sns: async (request) => {
+    if (!DELOYMENT_SNS_TOPIC) {
+      return {
+        status: 501,
+      };
+    }
+
+    try {
+      await validateSignature(request.request.body as Record<string, unknown>);
+    } catch {
+      console.error("SNS Signature validation failed");
+      return {
+        status: 400,
+      };
+    }
+
+    if (request.body.TopicArn != DELOYMENT_SNS_TOPIC) {
+      console.warn("Received request for unknown SNS topic");
+      return {
+        status: 400,
+      };
+    }
+
+    if (request.body.Type == "SubscriptionConfirmation" && request.body.Token) {
+      await confirmSubscription({
+        Token: request.body.Token,
+        TopicArn: DELOYMENT_SNS_TOPIC,
+      });
+    }
+
+    if (
+      request.body.Type == "Notification" &&
+      request.body.Subject == "AWS CloudFormation Notification" &&
+      request.body.Message
+    ) {
+      const message = parseCloudFormationMessage(request.body.Message);
+      const result = deploymentResultFromNotification(message);
+
+      console.log("Received deployment result from CloudFormation", result);
+      const status = result.pending
+        ? "uploading"
```

Comment:
I will replace the deployment state "uploading" with "pending" which I think is more accurate.